momento-sdk/src/main/java/momento/sdk/TokenClient.java

```diff
@@ -0,0 +1,217 @@
+package momento.sdk;
+
+import com.google.gson.Gson;
+import com.google.protobuf.ByteString;
+import grpc.permission_messages.CacheRole;
+import grpc.permission_messages.ExplicitPermissions;
+import grpc.permission_messages.Permissions;
+import grpc.permission_messages.PermissionsType;
+import grpc.permission_messages.TopicRole;
+import java.util.concurrent.CompletableFuture;
+import javax.annotation.Nonnull;
+import momento.sdk.auth.CredentialProvider;
+import momento.sdk.auth.accessControl.CacheItemSelector;
+import momento.sdk.auth.accessControl.CacheSelector;
+import momento.sdk.auth.accessControl.DisposableToken;
+import momento.sdk.auth.accessControl.DisposableTokenPermission;
+import momento.sdk.auth.accessControl.DisposableTokenScope;
+import momento.sdk.auth.accessControl.ExpiresIn;
+import momento.sdk.auth.accessControl.TopicSelector;
+import momento.sdk.exceptions.InvalidArgumentException;
+import momento.sdk.exceptions.UnknownException;
+import momento.sdk.responses.auth.GenerateDisposableTokenResponse;
+import momento.token._GenerateDisposableTokenRequest;
+import momento.token._GenerateDisposableTokenResponse;
+
+public class TokenClient {
+  private final CredentialProvider credentialProvider;
+  private final ScsTokenGrpcStubsManager tokenGrpcStubsManager;
+
+  public TokenClient(@Nonnull CredentialProvider credentialProvider) {
+    this.credentialProvider = credentialProvider;
+    this.tokenGrpcStubsManager = new ScsTokenGrpcStubsManager(credentialProvider);
+  }
+
+  private Permissions permissionsFromDisposableTokenScope(DisposableTokenScope scope) {
+    Permissions.Builder permissions = Permissions.newBuilder();
+    ExplicitPermissions.Builder explicitPermissions = ExplicitPermissions.newBuilder();
+
+    for (DisposableTokenPermission perm : scope.getPermissions()) {
+      PermissionsType grpcPerm = disposableTokenPermissionToGrpcPermission(perm);
+      explicitPermissions.addPermissions(grpcPerm);
+    }
+
+    permissions.setExplicit(explicitPermissions.build());
+    return permissions.build();
+  }
+
+  private PermissionsType disposableTokenPermissionToGrpcPermission(
+      DisposableTokenPermission permission) {
+    PermissionsType.Builder result = PermissionsType.newBuilder();
+    if (permission instanceof DisposableToken.CachePermission) {
+      PermissionsType.CachePermissions.Builder cachePerms =
+          PermissionsType.CachePermissions.newBuilder();
+      switch (((DisposableToken.CachePermission) permission).getRole()) {
```

Comment:
I am a bit of a java :noob:, does java enforce exhaustive switch statements? Do we need a default branch here?

Reply:
`
But exhaustiveness checks don't end there! While a default branch will always make a switch exhaustive, it isn't required - if the cases cover all possible values, e.g. of an enum, that suffices
`

[ref](https://nipafx.dev/java-switch/#:~:text=But%20exhaustiveness%20checks%20don%27t%20end%20there!%20While%20a%20default%20branch%20will%20always%20make%20a%20switch%20exhaustive%2C%20it%20isn%27t%20required%20%2D%20if%20the%20cases%20cover%20all%20possible%20values%2C%20e.g.%20of%20an%20enum%2C%20that%20suffices%3A)